gson/src/test/java/org/baeldung/gson/deserialization/HashMapDeserializationUnitTest.java

```diff
@@ -0,0 +1,87 @@
+package org.baeldung.gson.deserialization;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.gson.internal.LinkedTreeMap;
+import org.baeldung.gson.entities.Employee;
+import org.baeldung.gson.serialization.HashMapDeserializer;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.*;
+import com.google.gson.reflect.TypeToken;
+
+public class HashMapDeserializationUnitTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(HashMapDeserializationUnitTest.class);
+
+    @Test
+    public void whenUsingHashMapClass_thenShouldReturnMapWithDefaultClasses() {
+
+        String jsonString = "{'employee.name':'Bob','employee.salary':10000, 'employee.active':true, "
+                + "'employee':{'id':10, 'name': 'Bob Willis', 'address':'London'}}";
```

Comment:
in this line, you would only indent two spaces over from where the line above begins